java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/RuntimeImageBuildStep.java

```diff
@@ -24,41 +24,65 @@
 import com.google.cloud.runtimes.builder.config.domain.JdkServerLookup;
 import com.google.cloud.runtimes.builder.config.domain.RuntimeConfig;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
 public abstract class RuntimeImageBuildStep implements BuildStep {
 
   private final JdkServerLookup jdkServerLookup;
+  private final String compatImageName;
 
-  protected RuntimeImageBuildStep(JdkServerLookup jdkServerLookup) {
+  protected RuntimeImageBuildStep(JdkServerLookup jdkServerLookup, String compatImageName) {
     this.jdkServerLookup = jdkServerLookup;
+    this.compatImageName = compatImageName;
   }
 
   @Override
   public void run(BuildContext buildContext) throws BuildStepException {
-    buildContext.getDockerfile().appendLine("FROM " + getBaseRuntimeImage(buildContext));
+    String baseImage;
+    try {
+      baseImage = getBaseRuntimeImage(buildContext);
```

Comment:
There should be a log statement saying which base image was selected based on which artifact.